python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -572,15 +637,42 @@ async def do_offset(self, data):
         """
         self.assert_enabled('offset')
 
-        for axis in self.model.offset:
-            self.model.add_offset(axis, getattr(data, axis))
+        # Verify that the appropriate correction loop is running
+        if (data.x + data.y + data.z + data.u + data.v != 0.0) and not self.corrections['hexapod']:
+            raise RuntimeError('Hexapod correction is not enabled. Offsets cannot be applied. '
+                               'See the enableCorrection to close the correction loop.')
+        if (data.m1 != 0.0) and not self.corrections['m1']:
+            raise RuntimeError('M1 correction is not enabled. Offset cannot be applied. '
+                               'See the enableCorrection to close the correction loop.')
+        if (data.m2 != 0.0) and not self.corrections['m2']:
```

Comment:
fixed the comparison to zero, but elif doesn't work here since it's possible to supply corrections that apply to more than one if statement (e.g. a hexapod offset and an offset for m1).

Reply:
nevermind i take that back. It's just raising exceptions in the if statement so it doesn't matter...